library/src/main/java/com/google/android/material/motion/runtime/TargetScope.java

```diff
@@ -118,24 +133,42 @@ private int getDetailedState() {
     return state;
   }
 
+  private void removeNamedPlan(NamedPlan plan, String name, Performer.NamedPlanPerformance performer) {
+    if (performer != null) {
+      performer.removePlan(plan, name);
+      namedCache.remove(name);
+    }
+  }
+
+  private Performer.NamedPlanPerformance getNamedPerformer(NamedPlan plan, String name, Object target) {
+    Performer.NamedPlanPerformance namedPerformer = namedCache.get(name);
+    if (namedPerformer == null) {
+      // create it
```

Comment:
Not 100% sure if this is the correct thing to do here. Should we *always* create a performer here? Perhaps I should at least check to make sure that the `plan` parameter is not null before proceeding?